### PR TITLE
[WIP] Adding validation to the role creation.

### PR DESCRIPTION
**Pull request by RoySegall**
#71 follow up.

### CODE REVIEW

File: src/Entity/OgRole.php [hunk 1]
```diff
@@ -6,177 +6,240 @@
...
-  public function getGid() {
-    return $this->get('gid')->value;
+  protected $groupType;
+
+  /**
+   * @var string
+   *
+   * The group bundle. i.e: article, page
+   */
+  protected $groupBundle;
```
Comment (damiankloip) [inline src/Entity/OgRole.php:74]: I think we should keep all of these properties the same as they config_key storing them. No need for additional mapping keys then either.

File: src/Entity/OgRole.php [hunk 2]
```diff
@@ -6,177 +6,240 @@
...
+   * @var array
+   *
+   * List of permissions.
+   */
+  protected $permissions;
+
+  /**
+   * @return int
+   */
+  public function getId() {
```
Comment (damiankloip) [inline src/Entity/OgRole.php:93]: Why do we need this? All entities already have a id () method that is consistent.

File: src/Entity/OgRole.php [hunk 3]
```diff
@@ -6,177 +6,240 @@
...
+   * @var integer
    *
-   * @return $this
+   * The identifier of the role.
    */
-  public function setGid($gid) {
-    $this->set('gid', $gid);
-    return $this;
-  }
+  protected $id;
```
Comment (damiankloip) [inline src/Entity/OgRole.php:46]: Role ID is also likely not going to be an integer if we're using config entities.

File: src/Entity/OgRole.php [hunk 4]
```diff
@@ -6,177 +6,240 @@
...
+   * The user ID which the role assign to.
+   */
+  protected $uid;
+
+  /**
+   * @var array
+   *
+   * List of permissions.
+   */
+  protected $permissions;
```
Comment (damiankloip) [inline src/Entity/OgRole.php:88]: Could default this to an array here. Then also maybe `add/removePermission` methods would be useful?

File: src/Entity/OgRole.php [hunk 5]
```diff
@@ -6,177 +6,240 @@
...
 
   /**
-   * @var String
+   * @param array $permissions
    *
-   * Unique role name per group.
+   * @return OgRole
    */
-  protected $name;
+  public function setPermissions($permissions) {
```
Comment (damiankloip) [inline src/Entity/OgRole.php:210]: type hint array

**Comment by amitaibu** [comment]
What's the status here?

**Comment by RoySegall** [comment]
I'll prefer to close this for now. The API has changed rapidly and is still changing. Let's see that things settled down and i'll re tackle it.

**Pull request closed without merging**